### PR TITLE
Docs: Mention Git in the "Optional" section of "Running from Source"

### PR DESCRIPTION
**Pull request by NewSoupVi**
GIt is required to install the Zilliandomizer package.

Also, this is probably just nice to have.

### CODE REVIEW

**Review by ScipioWright — APPROVED** [review]

File: docs/running from source.md [hunk 1]
```diff
@@ -69,6 +69,20 @@ It should be dropped as "SNI" into the root folder of the project. Alternatively
...
 
 
+## Optional: Git
+
+[Git](https://git-scm.com) is required to install some of the packages that Archipelago depends on.
+It may be possible to run Archipelago from source without it, at your own risk.
+
+It is also generally recommended to have Git installed and understand how to use it, especially if you're thinking about contributing.
+Git will allow you to keep your copy of the repository updated without having to redownload the .zip manually.
+It lets you manage branches or forks of the repository with ease, and it will be all but required to submit your changes via a pull request.
```
Comment (black-sliver) [inline docs/running from source.md:79]: I'm not sure about this section and would personally just drop it. It's not in-depth enough for someone to understand what any of that means, but describing what git is or does would go beyond the scope of this doc. (Also for some people, especially for documentation, the pencil tool on github might be better suited.)
```suggestion
```

File: docs/running from source.md [hunk 2]
```diff
@@ -69,6 +69,20 @@ It should be dropped as "SNI" into the root folder of the project. Alternatively
...
+
+You can download the latest release of Git at [The downloads page on the Git website]https://git-scm.com/downloads).
+
+Beyond that, there are also graphical interfaces for Git that make it more accessible.
+For repositories on Github (such as this one), [Github Desktop](https://desktop.github.com) is one such option.
```
Comment (black-sliver) [inline docs/running from source.md:84]: Can we (also) suggest the git integration of PyCharm? Getting people to use PyCharm is good.

**Comment by ThePhar** [comment]
Is Git actually optional if Zillion Randomizer needs it? What happens if you try to run from source and git isn't found?

**Comment by black-sliver** [comment]
> Is Git actually optional if Zillion Randomizer needs it? What happens if you try to run from source and git isn't found?

It'll spit out an error but still generate as long as there is no Zillion yaml. The annoying part is that it'll retry each time you run any program.

On a side note: When suggesting `git+https://...` I did not consider this. I think github also allows direct zip download links for commits, maybe that would've been smarter.

**Comment by black-sliver** [comment]
Okay, looked it up and tried it. Using a github archive tarball url does not work with pip:
> Make sure you're either building from a fully intact git repository or PyPI tarballs. Most other sources (such as GitHub's tarballs, a git checkout without the .git folder) don't contain the necessary metadata and will not work.

File: docs/running from source.md [hunk 3]
```diff
@@ -69,6 +69,20 @@ It should be dropped as "SNI" into the root folder of the project. Alternatively
...
+## Optional: Git
+
+[Git](https://git-scm.com) is required to install some of the packages that Archipelago depends on.
+It may be possible to run Archipelago from source without it, at your own risk.
+
+It is also generally recommended to have Git installed and understand how to use it, especially if you're thinking about contributing.
+Git will allow you to keep your copy of the repository updated without having to redownload the .zip manually.
+It lets you manage branches or forks of the repository with ease, and it will be all but required to submit your changes via a pull request.
+
+You can download the latest release of Git at [The downloads page on the Git website]https://git-scm.com/downloads).
```
[the remark this answers is not in the document]
Comment (NewSoupVi) [reply docs/running from source.md]: My b, I did do that but then I made changes and made the age old mistake of "Eh it's fine I only changed [x] I don't need to test it again"

File: docs/running from source.md (hunk 2 above)
Comment (NewSoupVi) [reply docs/running from source.md:82]: I did the thing

**Review by black-sliver — APPROVED** [review]
Thanks!

**Pull request merged**